Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,785 @@
+        return ticket_view, {}, {}
+    elif response.get('errorDescription'):
+        raise DemistoException(f"Error while adding a new ticket.\n {response.get('errorDescription')}")
+    else:
+        raise DemistoException(f'Error while adding a new ticket.')
+
+
+def create_body_from_args(hd_queue_id: str = None, title: str = None, summary: str = None, impact: str = None,
+                          category: str = None, status: str = None, priority: str = None, machine: str = None,
+                          asset: str = None) -> dict:
+    """Function which creates the body of the request from user arguments.
+        Args:
+           hd_queue_id: the queue number to insert the ticket to.
+           title: title of the ticket.
+           summary: summary of the ticket.
+           impact: impact of the ticket.
+           category: category of the ticket.
+           status: status of the ticket.
+           priority: priority of the ticket.
+           machine: machine of the ticket.
+           asset: asset of the ticket.
+       Returns:
+           body of the request as a dict.
+    """
+    body = {}
+    if hd_queue_id:
+        body.update({'hd_queue_id': hd_queue_id})
+    if title:
+        body.update({'title': title})
+    if summary:
+        body.update({'summary': summary})
+    if impact:
+        body.update({'impact': impact})
+    if category:
+        body.update({'category': category})
+    if status:
+        body.update({'status': status})
+    if priority:
+        body.update({'priority': priority})
+    if machine:
+        body.update({'machine': machine})
+    if asset:
+        body.update({'asset': asset})
+    return body
+
+
+def update_ticket_command(client, args) -> Tuple[str, dict, dict]:
+    """Function which updates the body of the request from user arguments.
+        Args:
+            client : Integretion client which communicates with the api.
+            args: Users arguments of the command.
+       Returns:
+           human readable, context, raw response of this command.
+    """
+    ticket_id = args.get('ticket_id')
+    title = args.get("title")
+    summary = args.get('summary')
+    impact = args.get('impact')
+    category = args.get('category')
+    status = args.get('status')
+    priority = args.get('priority')
+    machine = args.get('machine')
+    asset = args.get('asset')
+    # owner = "for now"
+    custom_fields = args.get('custom_fields')
+
+    if not custom_fields:
+        temp_data = create_body_from_args(title=title, summary=summary, impact=impact, category=category, status=status,
+                                          priority=priority, machine=machine, asset=asset)
+        temp_data = {'Tickets': [temp_data]}
+    else:
+        temp_data = {'Tickets': [json.loads(custom_fields)]}
+    data = json.dumps(temp_data)
+
+    response = client.update_ticket_request(ticket_id, data)
+    if response.get('Result') == 'Success':
```

Comment:
Please reverse.
meaning if response was not successful raise and error, and then continue the logic(no need for `else: `)
Also, should not the context be updated?

Reply:
For now, the user will need to run tickets list command again after update to see the update. I can update context but only for params we enable and not for the custom_fields, as the user can change any key. Do you think I should update context for the known keys?